index.js

```diff
@@ -134,21 +134,55 @@ exports.createServer = function (opts) {
       ogr.timeout(opts.timeout)
     }
 
-    let format = req.body.format || 'shp'
+    let format = req.body.format.toLowerCase() || 'shp'
+
+    ogr.format(format)
+
+    const sendResponse = (buf) => {
 
-    ogr.format(format).exec(function (er, buf) {
-      if (isOgreFailureError(er))
-        return res
-          .status(400)
-          .json({errors: er.message.replace('\n\n', '').split('\n')})
-      if (er) return next(er)
       res.header('Content-Type', 'application/zip')
       res.header(
         'Content-Disposition',
         'filename=' + (req.body.outputName || 'ogre.zip')
       )
       res.end(buf)
-    })
+
+    }
+
+    try {
+
+      switch (format) {
+        // These formats must use .destination
+        case 'dxf':
+        case 'dgn':
+        case 'txt':
+        case 'gxt':
+        case 'gmt':
+          
+          // Random string to prevent multiple request being overwritten
+          let randomId = Math.random().toString(36).substring(7);
+
+          let tmpDestination = join(__dirname, `/tmp-${randomId}.${format}`)
```

Comment:
Can this use the OS temp directory please?

Reply:
Yes